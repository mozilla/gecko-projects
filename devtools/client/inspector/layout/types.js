/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { PropTypes } = require("devtools/client/shared/vendor/react");

/**
 * A single grid container in the document.
 */
exports.grid = {

  // The id of the grid
  id: PropTypes.number,

  // The node front of the grid container
  nodeFront: PropTypes.object,

  // The grid fragment object of the grid container
  gridFragments: PropTypes.object

};
