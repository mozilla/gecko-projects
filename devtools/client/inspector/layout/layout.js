/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

function LayoutViewTool(inspector, window) {
  this.inspector = inspector;
  this.document = window.document;
  this.store = null;

  this.init();
}

LayoutViewTool.prototype = {

  init() {
    const { React, ReactDOM, ReactRedux, browserRequire } = this.inspector;

    const Store = browserRequire("devtools/client/inspector/layout/store");
    const App = React.createFactory(
      browserRequire("devtools/client/inspector/layout/components/App"));

    let store = this.store = Store();
    let provider = React.createElement(ReactRedux.Provider, { store }, App());
    ReactDOM.render(provider, this.document.querySelector("#layoutview-container"));
  },

  destroy() {
    this.inspector = null;
    this.document = null;
    this.store = null;
  },

};

exports.LayoutViewTool = LayoutViewTool;
