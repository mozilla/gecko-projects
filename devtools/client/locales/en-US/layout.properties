# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

# LOCALIZATION NOTE This file contains the Layout Inspector strings.
# The Layout Inspector is a panel accessible in the Inspector sidebar.

# LOCALIZATION NOTE (flexbox.header): The accordion header for the Flexbox panel when
# no flex container or item is selected.
flexbox.header=Flexbox

# LOCALIZATION (flexbox.flexContainer): The accordion header for the Flexbox panel
# when a flex container is selected.
flexbox.flexContainer=Flex Container

# LOCALIZATION NOTE) (flexbox.flexItemOf): The accordion header for the Flexbox panel
# when a flex item is selected. %s represents the flex container selector.
flexbox.flexItemOf=Flex Item of %S

# LOCALIZATION NOTE (flexbox.noFlexboxeOnThisPage): In the case where there are no CSS
# flex containers to display.
flexbox.noFlexboxeOnThisPage=Select a Flex container or item to continue.

# LOCALIZATION NOTE (flexbox.flexContainerProperties): Header for the flex container
# properties in the Flexbox panel.
flexbox.flexContainerProperties=Flex Container Properties

# LOCALIZATION NOTE (flexbox.contentWidth, flexbox.contentHeight, flexbox.finalWidth,
# flexbox.finalHeight): Labels for the flex item sizing properties in the Flexbox panel.
flexbox.contentWidth=Content width:
flexbox.contentHeight=Content height:
flexbox.finalWidth=Final width:
flexbox.finalHeight=Final height:

# LOCALIZATION NOTE (flexbox.itemSizing.baseSizeSectionHeader): Header label displayed
# at the start of the flex item sizing Base Size section.
flexbox.itemSizing.baseSizeSectionHeader=Base Size

# LOCALIZATION NOTE (flexbox.itemSizing.flexibilitySectionHeader): Header label displayed
# at the start of the flex item sizing Flexibility section.
flexbox.itemSizing.flexibilitySectionHeader=Flexibility

# LOCALIZATION NOTE (flexbox.itemSizing.minSizeSectionHeader): Header label displayed
# at the start of the flex item sizing Minimum Size section.
flexbox.itemSizing.minSizeSectionHeader=Minimum Size

# LOCALIZATION NOTE (flexbox.itemSizing.maxSizeSectionHeader): Header label displayed at
# the start of the flex item sizing Maximum Size section.
flexbox.itemSizing.maxSizeSectionHeader=Maximum Size

# LOCALIZATION NOTE (flexbox.itemSizing.finalSizeSectionHeader): Header label displayed at
# the start of the flex item sizing Final Size section.
flexbox.itemSizing.finalSizeSectionHeader=Final Size

# LOCALIZATION NOTE (flexbox.itemSizing.itemBaseSizeFromContent): Label shown in the flex
# item sizing panel. It tells users that a given item’s base size was calculated from its
# content size when unconstrained.
flexbox.itemSizing.itemBaseSizeFromContent=The item’s content size when unconstrained.

# LOCALIZATION NOTE (flexbox.itemSizing.itemMinSizeFromItemMinContent): Label shown in the
# flex item sizing panel. It tells users that a given item’s minimum size is coming from
# its min-content size.
flexbox.itemSizing.itemMinSizeFromItemMinContent=This is the element’s minimum content size.

# LOCALIZATION NOTE (flexbox.itemSizing.extraRoomOnLine): Label shown in the flexbox item
# sizing panel. It tells users that there was extra room to distribute inside a given flex
# line.
flexbox.itemSizing.extraRoomOnLine=There was extra room available on the flex line.

# LOCALIZATION NOTE (flexbox.itemSizing.notEnoughRoomOnLine): Label shown in the flexbox
# item sizing panel. It tells users that there wasn’t enough room inside a given flex line
# for all of its items.
flexbox.itemSizing.notEnoughRoomOnLine=There wasn’t enough room available on the flex line.

# LOCALIZATION NOTE (flexbox.itemSizing.growthAttemptWhenClamped): Label shown in the
# flexbox item sizing panel. It tells users that a given item attempted to grow by a
# certain amount but ended up being clamped by a max size.
# (note that clamp is a common word in flexbox terminology. It refers to constraining an
# item's size to some defined min/max-width/height set on the element, even though there
# might have been room for it to grow, or reason for it to shrink more).
flexbox.itemSizing.growthAttemptWhenClamped=The item wanted to grow, but it was clamped.

# LOCALIZATION NOTE (flexbox.itemSizing.shrinkAttemptWhenClamped): Label shown in the
# flexbox item sizing panel. It tells users that a given item attempted to shrink by a
# certain amount but ended up being clamped by a min size.
flexbox.itemSizing.shrinkAttemptWhenClamped=The item wanted to shrink, but it was clamped.

# LOCALIZATION NOTE (flexbox.itemSizing.shrinkAttemptButCouldnt): Label shown in the
# flexbox item sizing panel. It tells users that a given item attempted to shrink by a
# certain amount but could not
flexbox.itemSizing.shrinkAttemptButCouldnt=Item was set to shrink but could not.

# LOCALIZATION NOTE (flexbox.itemSizing.growthAttemptButSiblings): Label shown in the
# flexbox item sizing panel. It tells users that a given item could not grow to occupy
# extra space because its siblings have likely already used it.
flexbox.itemSizing.growthAttemptButSiblings=Item could not grow, siblings have likely used the extra space.

# LOCALIZATION NOTE (flexbox.itemSizing.setToGrow): Label shown in the flex item sizing
# panel. It tells users that a given item was set to grow.
flexbox.itemSizing.setToGrow=Item was set to grow.

# LOCALIZATION NOTE (flexbox.itemSizing.setToShrink): Label shown in the flexbox item
# sizing panel. It tells users that a given item was set to shrink.
flexbox.itemSizing.setToShrink=Item was set to shrink.

# LOCALIZATION NOTE (flexbox.itemSizing.notSetToGrow): Label shown in the
# flexbox item sizing panel. It tells users that a given item was not set to grow, even
# though there might have been space on the flex line for it to grow.
flexbox.itemSizing.notSetToGrow=Item was not set to grow.

# LOCALIZATION NOTE (flexbox.itemSizing.notSetToShrink): Label shown in the
# flexbox item sizing panel. It tells users that a given item did not shrink even though
# there might not have been enough space on the flex line for all items to fit.
flexbox.itemSizing.notSetToShrink=Item was not set to shrink.

# LOCALIZATION NOTE (layout.cannotShowGridOutline, layout.cannotSHowGridOutline.title):
# In the case where the grid outline cannot be effectively displayed.
layout.cannotShowGridOutline=Cannot show outline for this grid
layout.cannotShowGridOutline.title=The selected grid’s outline cannot effectively fit inside the layout panel for it to be usable.

# LOCALIZATION NOTE (layout.displayAreaNames): Label of the display area names setting
# option in the CSS Grid panel.
layout.displayAreaNames=Display area names

# LOCALIZATION NOTE (layout.displayLineNumbers): Label of the display line numbers
# setting option in the CSS Grid panel.
layout.displayLineNumbers=Display line numbers

# LOCALIZATION NOTE (layout.extendLinesInfinitely): Label of the extend lines
# infinitely setting option in the CSS Grid panel.
layout.extendLinesInfinitely=Extend lines infinitely

# LOCALIZATION NOTE (layout.header): The accordion header for the CSS Grid panel.
layout.header=Grid

# LOCALIZATION NOTE (layout.gridDisplaySettings): The header for the grid display
# settings container in the CSS Grid panel.
layout.gridDisplaySettings=Grid Display Settings

# LOCALIZATION NOTE (layout.noGridsOnThisPage): In the case where there are no CSS grid
# containers to display.
layout.noGridsOnThisPage=CSS Grid is not in use on this page

# LOCALIZATION NOTE (layout.overlayMultipleGrids): The header for the list of grid
# container elements that can be highlighted in the CSS Grid panel.
layout.overlayMultipleGrids=Overlay Multiple Grids

# LOCALIZATION NOTE (layout.overlayGrid): Alternate header for the list of grid container
# elements if only one item can be selected.
layout.overlayGrid=Overlay Grid
