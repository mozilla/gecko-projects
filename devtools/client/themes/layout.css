/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

#sidebar-panel-layoutview {
  margin: 0;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

#layoutview-container {
  height: 100%;
  overflow: auto;
}

#layoutview-container-focusable {
  height: 100%;
  outline: none;
}

.layoutview-no-grids {
  font-style: italic;
  text-align: center;
  padding: 0.5em;
}
