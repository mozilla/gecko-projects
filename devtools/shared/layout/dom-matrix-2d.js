/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

/**
 * Returns a matrix for the scaling given.
 * Calling `scale()` or `scale(1) returns a new identity matrix.
 *
 * @param {Number} [sx = 1]
 *        the abscissa of the scaling vector.
 *        If unspecified, it will equal to `1`.
 * @param {Number} [sy = sx]
 *        The ordinate of the scaling vector.
 *        If not present, its default value is `sx`, leading to a uniform scaling.
 * @return {Array}
 *         The new matrix.
 */
const scale = (sx = 1, sy = sx) => [
  sx, 0, 0,
  0, sy, 0,
  0, 0, 1
];
exports.scale = scale;

/**
 * Returns a matrix for the translation given.
 * Calling `translate()` or `translate(0) returns a new identity matrix.
 *
 * @param {Number} [tx = 0]
 *        The abscissa of the translating vector.
 *        If unspecified, it will equal to `0`.
 * @param {Number} [ty = tx]
 *        The ordinate of the translating vector.
 *        If unspecified, it will equal to `tx`.
 * @return {Array}
 *         The new matrix.
 */
const translate = (tx = 0, ty = tx) => [
  1, 0, tx,
  0, 1, ty,
  0, 0, 1
];
exports.translate = translate;

/**
 * Returns a new identity matrix.
 *
 * @return {Array}
 *         The new matrix.
 */
const identity = () => [
  1, 0, 0,
  0, 1, 0,
  0, 0, 1
];
exports.identity = identity;

/**
 * Multiplies two matrices and returns a new matrix with the result.
 *
 * @param {Array} M1
 *        The first operand.
 * @param {Array} M2
 *        The second operand.
 * @return {Array}
 *        The resulting matrix.
 */
const multiply = (M1, M2) => {
  let c11 = M1[0] * M2[0] + M1[1] * M2[3] + M1[2] * M2[6];
  let c12 = M1[0] * M2[1] + M1[1] * M2[4] + M1[2] * M2[7];
  let c13 = M1[0] * M2[2] + M1[1] * M2[5] + M1[2] * M2[8];

  let c21 = M1[3] * M2[0] + M1[4] * M2[3] + M1[5] * M2[6];
  let c22 = M1[3] * M2[1] + M1[4] * M2[4] + M1[5] * M2[7];
  let c23 = M1[3] * M2[2] + M1[4] * M2[5] + M1[5] * M2[8];

  let c31 = M1[6] * M2[0] + M1[7] * M2[3] + M1[8] * M2[6];
  let c32 = M1[6] * M2[1] + M1[7] * M2[4] + M1[8] * M2[7];
  let c33 = M1[6] * M2[2] + M1[7] * M2[5] + M1[8] * M2[8];

  return [
    c11, c12, c13,
    c21, c22, c23,
    c31, c32, c33
  ];
};
exports.multiply = multiply;

/**
 * Applies the given matrix to a point.
 *
 * @param {Array} M
 *        The matrix to apply.
 * @param {Array} P
 *        The point's vector.
 * @return {Array}
 *        The resulting point's vector.
 */
const apply = (M, P) => [
  M[0] * P[0] + M[1] * P[1] + M[2],
  M[3] * P[0] + M[4] * P[1] + M[5],
];
exports.apply = apply;

/**
 * Returns the transformation origin point for the given node.
 *
 * @param {DOMNode} node
 *        The node.
 * @return {Array}
 *        The transformation origin point.
 */
function getNodeTransformOrigin(node) {
  let origin = node.ownerGlobal.getComputedStyle(node).transformOrigin;

  return origin.split(/ /).map(parseFloat);
}
exports.getNodeTransformOrigin = getNodeTransformOrigin;

/**
 * Returns the transformation matrix for the given node.
 *
 * @param {DOMNode} node
 *        The node.
 * @return {Array}
 *        The transformation matrix.
 */
function getNodeTransformationMatrix(node) {
  let t = node.ownerGlobal.getComputedStyle(node).transform;

  if (t === "none") {
    return null;
  }

  // We're assuming is a 2d matrix.
  let m = t.substring(t.indexOf("(") + 1, t.length - 1).split(/,\s*/).map(Number);
  let [a, b, c, d, e, f] = m;

  // If the length is 16, it's a 3d matrix: in that case we'll extrapolate only the values
  // we need for the 2D transformation; this cover the scenario where 3D CSS properties
  // are used only for HW acceleration on 2D transformation.
  if (m.length === 16) {
    c = m[4];
    d = m[5];
    e = m[12];
    f = m[13];
  }

  return [
    a, c, e,
    b, d, f,
    0, 0, 1
  ];
}

exports.getNodeTransformationMatrix = getNodeTransformationMatrix;
