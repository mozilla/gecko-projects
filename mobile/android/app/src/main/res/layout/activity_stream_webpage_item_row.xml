<?xml version="1.0" encoding="utf-8"?>
<!-- This Source Code Form is subject to the terms of the Mozilla Public
   - License, v. 2.0. If a copy of the MPL was not distributed with this
   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:gecko="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:background="?android:attr/selectableItemBackground"
    android:orientation="horizontal">

    <!-- On some devices, marginBottom is ignored for Views within a RelativeLayout.
         Wrapping in a FrameLayout appears to be a sufficient workaround for this issue -->
    <FrameLayout
        android:id="@+id/icon_wrapper"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content">

        <org.mozilla.gecko.activitystream.homepanel.stream.StreamOverridablePageIconLayout
                android:id="@+id/page_icon"
                android:layout_width="@dimen/favicon_bg"
                android:layout_height="@dimen/favicon_bg"
                android:layout_marginStart="@dimen/activity_stream_base_margin"
                android:layout_marginLeft="@dimen/activity_stream_base_margin"
                android:layout_marginTop="@dimen/activity_stream_base_margin"
                android:layout_marginBottom="@dimen/activity_stream_base_margin"
                tools:background="@drawable/favicon_globe" />

    </FrameLayout>

    <ImageView
        android:id="@+id/menu"
        android:layout_width="32dp"
        android:layout_height="48dp"
        android:layout_alignParentEnd="true"
        android:layout_alignParentRight="true"
        android:layout_alignParentTop="true"
        android:layout_gravity="right|end|top"
        android:contentDescription="@string/menu"
        android:paddingBottom="16dp"
        android:paddingTop="16dp" />

    <!-- In RTL locales, we don't want a large horizontal gap between the domain, page title and source
         text so we set textAlignment=viewStart for all of them so that it's to the left in LTR and
         right in RTL. Note: to appease the RtlCompat lint warning, gravity="start" is also required. -->
    <TextView
        android:id="@+id/page_domain"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_toEndOf="@id/icon_wrapper"
        android:layout_toLeftOf="@id/menu"
        android:layout_toRightOf="@id/icon_wrapper"
        android:layout_toStartOf="@id/menu"
        android:paddingLeft="@dimen/activity_stream_base_margin"
        android:paddingStart="@dimen/activity_stream_base_margin"
        android:paddingRight="0dp"
        android:paddingEnd="0dp"
        android:paddingTop="@dimen/activity_stream_base_margin"
        android:textAlignment="viewStart"
        android:gravity="start"
        android:textColor="@color/activity_stream_subtitle"
        android:textSize="12sp"
        tools:text="twitter" />

    <!-- See id/page_domain for textAlignment comment. -->
    <TextView
        android:id="@+id/page_title"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_below="@id/page_domain"
        android:layout_toEndOf="@id/icon_wrapper"
        android:layout_toLeftOf="@id/menu"
        android:layout_toRightOf="@id/icon_wrapper"
        android:layout_toStartOf="@id/menu"
        android:ellipsize="end"
        android:maxLines="3"
        android:paddingLeft="@dimen/activity_stream_base_margin"
        android:paddingStart="@dimen/activity_stream_base_margin"
        android:paddingRight="0dp"
        android:paddingEnd="0dp"
        android:textAlignment="viewStart"
        android:gravity="start"
        android:textColor="#ff000000"
        android:textSize="14sp"
        android:textStyle="bold"
        tools:text="Descriptive title of a page that is veeeeeeery long - maybe even too long?" />

    <LinearLayout
        android:id="@+id/page_source_layout"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_below="@id/page_title"
        android:layout_toEndOf="@id/icon_wrapper"
        android:layout_toRightOf="@id/icon_wrapper"
        android:gravity="center_vertical"
        android:orientation="horizontal"
        android:paddingEnd="@dimen/activity_stream_base_margin"
        android:paddingLeft="@dimen/activity_stream_base_margin"
        android:paddingRight="@dimen/activity_stream_base_margin"
        android:paddingStart="@dimen/activity_stream_base_margin"
        android:paddingTop="4dp"
        tools:ignore="UseCompoundDrawables">

        <ImageView
            android:id="@+id/page_source_icon"
            android:layout_width="12dp"
            android:layout_height="12dp"
            tools:src="@drawable/ic_as_bookmarked" />

        <!-- See id/page_domain for textAlignment comment. -->
        <TextView
            android:id="@+id/page_source"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_marginStart="2dp"
            android:layout_marginLeft="2dp"
            android:layout_marginRight="0dp"
            android:layout_marginEnd="0dp"
            android:layout_weight="1"
            android:textAlignment="viewStart"
            android:gravity="start"
            android:textColor="@color/activity_stream_subtitle"
            android:textSize="12sp"
            tools:text="Bookmarked" />

    </LinearLayout>

    <TextView
        android:id="@+id/switch_tab_hint"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_below="@id/page_source_layout"
        android:layout_toEndOf="@id/icon_wrapper"
        android:layout_toRightOf="@id/icon_wrapper"
        android:drawableStart="@drawable/ic_url_bar_tab"
        android:drawableLeft="@drawable/ic_url_bar_tab"
        android:paddingStart="6dp"
        android:paddingLeft="6dp"
        android:paddingEnd="0dp"
        android:paddingRight="0dp"
        android:text="@string/switch_to_tab"
        android:textColor="@color/activity_stream_subtitle"
        android:textSize="12sp"
        tools:drawableLeft="@drawable/ic_url_bar_tab"
        tools:drawableStart="@drawable/ic_url_bar_tab"
        tools:text="@string/switch_to_tab" />

    <!-- Empty bottom space added irrespective of "switch_tab_hint" or "page_source_layout"
         being the bottom view. ("switch_tab_hint" can have visibility="gone") -->
    <View
        android:id="@+id/layout_bottom_margin"
        android:layout_width="match_parent"
        android:layout_height="@dimen/activity_stream_base_margin"
        android:layout_alignParentBottom="true"
        android:layout_below="@id/switch_tab_hint" />

</RelativeLayout>
