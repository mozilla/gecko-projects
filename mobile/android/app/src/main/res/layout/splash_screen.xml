<?xml version="1.0" encoding="utf-8"?>
<org.mozilla.gecko.widget.SplashScreen xmlns:android="http://schemas.android.com/apk/res/android"
                                       xmlns:app="http://schemas.android.com/apk/res-auto"
                                       android:id="@+id/splash_root"
                                       android:layout_width="match_parent"
                                       android:layout_height="match_parent"
                                       android:background="#ffffff"
                                       android:orientation="vertical">

    <ImageView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_centerInParent="true"
        android:src="@drawable/firefox_logo"/>

</org.mozilla.gecko.widget.SplashScreen>