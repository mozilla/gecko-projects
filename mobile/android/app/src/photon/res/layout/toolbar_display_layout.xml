<?xml version="1.0" encoding="utf-8"?>

<!-- This Source Code Form is subject to the terms of the Mozilla Public
   - License, v. 2.0. If a copy of the MPL was not distributed with this
   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->

<merge xmlns:android="http://schemas.android.com/apk/res/android"
       xmlns:gecko="http://schemas.android.com/apk/res-auto"
       xmlns:tools="http://schemas.android.com/tools"
       tools:background="@android:color/black"
       tools:layout_height="@dimen/browser_toolbar_height"
       tools:layout_width="match_parent"
       tools:parentTag="LinearLayout">

    <!-- The site security icon is misaligned with the page title so
         we add a bottom margin to align their bottoms.
         Site security icon must have exact position and size as search icon in
         edit layout -->

    <org.mozilla.gecko.widget.themed.ThemedImageButton
        android:id="@+id/site_security"
        style="@style/UrlBar.SiteIdentity"
        android:layout_gravity="center_vertical"
        android:background="@drawable/url_bar_title_bg"
        android:contentDescription="@string/site_security"
        android:src="@drawable/security_mode_icon"
        tools:src="@drawable/ic_lock"/>

    <org.mozilla.gecko.widget.FadedMultiColorTextView
        android:id="@+id/url_bar_title"
        style="@style/UrlBar.Title"
        android:layout_width="match_parent"
        android:layout_gravity="center_vertical"
        android:layout_weight="1.0"
        android:background="@drawable/url_bar_title_bg"
        gecko:fadeBackgroundColor="@android:color/transparent"
        gecko:fadeWidth="40dip"/>

    <org.mozilla.gecko.toolbar.PageActionLayout
        android:id="@+id/page_action_layout"
        android:layout_width="wrap_content"
        android:layout_height="match_parent"
        android:orientation="horizontal"
        android:visibility="gone"
        tools:visibility="visible"/>

    <org.mozilla.gecko.toolbar.ToolbarRoundButton
        android:id="@+id/stop"
        style="@style/UrlBar.ImageButton"
        android:background="@drawable/url_bar_action_button"
        android:contentDescription="@string/stop"
        android:src="@drawable/ic_cancel"
        android:visibility="gone"
        tools:visibility="visible"/>

</merge>
