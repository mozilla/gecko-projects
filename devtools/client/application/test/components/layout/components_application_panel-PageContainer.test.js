/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

// Import libs
const { shallow } = require("enzyme");
const { createFactory } = require("react");

// Import setupStore with imported & combined reducers
const {
  setupStore,
} = require("devtools/client/application/test/components/helpers/helpers");

const PageContainer = createFactory(
  require("devtools/client/application/src/components/layout/PageContainer")
);

const { PAGE_TYPES } = require("devtools/client/application/src/constants");

/**
 * Test for workerListEmpty.js component
 */

describe("PageContainer", () => {
  function buildStoreWithSelectedPage(selectedPage) {
    return setupStore({
      preloadedState: {
        ui: {
          selectedPage,
        },
      },
    });
  }

  it("renders the WorkersPage component when workers page is selected", () => {
    const store = buildStoreWithSelectedPage(PAGE_TYPES.SERVICE_WORKERS);
    const wrapper = shallow(PageContainer({ store })).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders nothing when no page is selected", () => {
    const store = buildStoreWithSelectedPage(null);
    const wrapper = shallow(PageContainer({ store })).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders nothing when an invalid page is selected", () => {
    const store = buildStoreWithSelectedPage("foo");
    const wrapper = shallow(PageContainer({ store })).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
