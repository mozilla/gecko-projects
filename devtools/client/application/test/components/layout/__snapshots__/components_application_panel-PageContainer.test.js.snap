// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer renders nothing when an invalid page is selected 1`] = `""`;

exports[`PageContainer renders nothing when no page is selected 1`] = `""`;

exports[`PageContainer renders the WorkersPage component when workers page is selected 1`] = `<Connect(WorkersPage) />`;
